### PR TITLE
wal: fix crc mismatch crash bug

### LINKED ISSUE #11918: 3.3.21 crashing on every node with "etcdmain: walpb: crc mismatch"
I am attempting to do a rolling upgrade from 3.3.20 to 3.3.21, and every node has failed with "etcdmain: walpb: crc mismatch". For the first node, I removed it and re-bootstrapped it successfully, and it can stop and start fine on 3.3.21 after re-bootstrapping, but now that the second node in the cluster is crashing on startup with the same error, I'm a little suspicious that this is actually a bug in 3.3.21.

None of these machines have ever suffered any hardware failure or unexpected shutdown, and I have been upgrading them to every 3.3.x patch release without ever seeing this issue before.

These nodes are mostly used with V2-protocol-speaking applications, so almost all the data is in the V2 store. Not sure if that makes a difference.

### PR DESCRIPTION
**Pull request by tangcong**
fix #11918

### CODE REVIEW

**Comment by tangcong** [comment]
```
=== RUN   TestValidSnapshotEntriesAfterPurgeWal
{"level":"info","msg":"created a new WAL segment","path":"/var/folders/g4/c6b5qg7x05z52phyxcb6wtlm0000gn/T/waltest889121245/0000000000000001-0000000000000002.wal"}
{"level":"info","msg":"created a new WAL segment","path":"/var/folders/g4/c6b5qg7x05z52phyxcb6wtlm0000gn/T/waltest889121245/0000000000000002-0000000000000004.wal"}
    TestValidSnapshotEntriesAfterPurgeWal: wal_test.go:1105: walpb: crc mismatch
--- FAIL: TestValidSnapshotEntriesAfterPurgeWal (0.08s)
FAIL
```
@gyuho @jpbetz  i added a testcase to reproduce it,it is easy to reproduce it.  could we need to release a new 3.4/3.3 version ?

File: CHANGELOG-3.5.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ Note that any `etcd_debugging_*` metrics are experimental and subject to change.
   - Previously, server restore fails if it had crashed after persisting raft hard state but before saving snapshot.
   - See https://github.com/etcd-io/etcd/issues/10219 for more.
 - Improve logging around snapshot send and receive.
+- Fix [crc mismatch crash bug](https://github.com/etcd-io/etcd/pull/11924).
```
Comment (gyuho) [inline CHANGELOG-3.5.md:122]: `Add missing CRC checksum check in WAL validate method otherwise causes panic https://github.com/etcd-io/etcd/issues/11918`?

**Comment by gyuho** [comment]
lgtm, I will backport with new release. Thanks for the fix!

/cc @jpbetz

File: CHANGELOG-3.5.md (hunk 1 above)
Comment (tangcong) [reply CHANGELOG-3.5.md]: updated. thanks. @gyuho

**Comment by jpbetz** [comment]
Yikes. This is my fault. Thanks for fixing.

lgtm

**Pull request merged**